lib/redlock/client.rb

```diff
@@ -30,13 +30,27 @@ def initialize(servers = DEFAULT_REDIS_URLS, options = {})
       @retry_delay = options[:retry_delay] || DEFAULT_RETRY_DELAY
     end
 
+    def testing=(mode)
+      @testing_mode = mode
+    end
+
     # Locks a resource for a given time.
     # Params:
     # +resource+:: the resource (or key) string to be locked.
     # +ttl+:: The time-to-live in ms for the lock.
     # +block+:: an optional block that automatically unlocks the lock.
     def lock(resource, ttl, &block)
-      lock_info = try_lock_instances(resource, ttl)
+      if @testing_mode == :bypass
+        lock_info = {
+          validity: ttl,
+          resource: resource,
+          value: SecureRandom.uuid
+        }
+      elsif @testing_mode == :fail
+        return false
```

Comment:
@ryansch With this line you modified the behaviour of `lock()` to not `yield` in case `try_lock_instances` failed. In fact, your application code should probably have error handling for when the block parameter is `false`, so I presume you would want to test that, too.

Change it to `lock_info = false`?